Mengimport package pandas

In [1]:
import pandas as pd
import numpy as np

**NOMOR 1**

Membuat dan menampilkan DataFrame dengan nama "df" yang berisi data pada soal

In [2]:
df = pd.DataFrame({
    "name": ["Anna", "Dane", "David", "Kevin", "Joe", "Rose"],
    "weight": [45.2, 50.5, 62.1, 64.0, 59.0, np.nan],
    "height": [155, 160, 162, 170, 167, 159],
    "birth_year": [2000,1997,1996,1987,1998,2003],
    "gender": ["female", "male", "male", "male", "male", "female"],
    "city": ["New York", "Chicago", "Austin", "New York", np.nan, "New York"]
    })
df

name  weight  height  birth_year  gender      city
0   Anna    45.2     155        2000  female  New York
1   Dane    50.5     160        1997    male   Chicago
2  David    62.1     162        1996    male    Austin
3  Kevin    64.0     170        1987    male  New York
4    Joe    59.0     167        1998    male       NaN
5   Rose     NaN     159        2003  female  New York

**NOMOR 2**

Menampilkan basic DataFrame info dari "df"

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6 entries, 0 to 5
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   name        6 non-null      object 
 1   weight      5 non-null      float64
 2   height      6 non-null      int64  
 3   birth_year  6 non-null      int64  
 4   gender      6 non-null      object 
 5   city        5 non-null      object 
dtypes: float64(1), int64(2), object(3)
memory usage: 416.0+ bytes


Interpretasi 
Pada DataFrame "df" memiliki jumlah 6 kolom, dimana :
1. Kolom "name" memiliki 6 data yang terisi, tipe data object
2. Kolom "weight" memiliki 5 data yang terisi, tipe data float64
3. Kolom "height" memiliki 6 data yang terisi, tipe data int64
4. Kolom "birth_year" memiliki 6 data yang terisi, tipe data int64
5. Kolom "gender" memiliki 6 data yang terisi, tipe data object
6. Kolom "city" memiliki 5 data yang terisi, tipe data object



**NOMOR 3**

Menampilkan  basic statistic information for numerical columns

In [4]:
df.describe()

weight      height   birth_year
count   5.000000    6.000000     6.000000
mean   56.160000  162.166667  1996.833333
std     8.015173    5.492419     5.419102
min    45.200000  155.000000  1987.000000
25%    50.500000  159.250000  1996.250000
50%    59.000000  161.000000  1997.500000
75%    62.100000  165.750000  1999.500000
max    64.000000  170.000000  2003.000000

Terdapat 3 data numerical pada DataFrame "df" 
Interpretasi statistik deskripti berdasarkan output diatas, sebagai berikut :
1. Variabel weight memiliki 5 data, height 6 data, dan birth_year 6 data
2. Rata-rata berat badan sebesar 56.16 kg, rata-rata tinggi badan sebesar 162.17 cm.
3. standar deviasi weight sebesar 8.02, height 5.49, dan birth_year 5.42
4. Data terendah/minimum variabel weight sebesar 45.2 kg, height sebesar 155 cm, dan birth_year 1987
5. Sedangkan data tertinggi/maksimum variabel weight sebesar 64 kg, height sebesar 170 cm, dan birth_year 2003


**NOMOR 4**

Menyusun ulang kolom agar sesuai dengan Pandas DataFrame pada soal

In [5]:
df = df.reindex(columns=["name","birth_year","height","weight","city","gender"])
df

name  birth_year  height  weight      city  gender
0   Anna        2000     155    45.2  New York  female
1   Dane        1997     160    50.5   Chicago    male
2  David        1996     162    62.1    Austin    male
3  Kevin        1987     170    64.0  New York    male
4    Joe        1998     167    59.0       NaN    male
5   Rose        2003     159     NaN  New York  female

**NOMOR 5**

a) Menampilkan data df dengan kondisi seseorang berjenis kelamin laki-laki dan lahir setelah tahun 1996

In [6]:
df[(df["gender"] == "male") & (df["birth_year"]>1996)]  

name  birth_year  height  weight     city gender
1  Dane        1997     160    50.5  Chicago   male
4   Joe        1998     167    59.0      NaN   male

b) Menampilkan data df dengan kondisi seseorang berjenis kelamin perempuan dan memiliki tinggi lebih dari 157

In [7]:
df[(df["gender"] == "female") & (df["height"]>157)]  

name  birth_year  height  weight      city  gender
5  Rose        2003     159     NaN  New York  female

**NOMOR 6**


In [12]:
df["weight"] = df["weight"].fillna(df[df['gender'] == 'female']['weight'].mean())
df

name  birth_year  height  weight      city  gender
0   Anna        2000     155    45.2  New York  female
1   Dane        1997     160    50.5   Chicago    male
2  David        1996     162    62.1    Austin    male
3  Kevin        1987     170    64.0  New York    male
4    Joe        1998     167    59.0       NaN    male
5   Rose        2003     159    45.2  New York  female

Mengisi missing value (NaN) city (weight) dengan modus dari variabel city

In [13]:
df["city"] = df["city"].fillna(df["city"].mode()[0])
df

name  birth_year  height  weight      city  gender
0   Anna        2000     155    45.2  New York  female
1   Dane        1997     160    50.5   Chicago    male
2  David        1996     162    62.1    Austin    male
3  Kevin        1987     170    64.0  New York    male
4    Joe        1998     167    59.0  New York    male
5   Rose        2003     159    58.9  New York  female

Mengimport package scipy.stats

In [14]:
import scipy.stats as stats

**NOMOR 7**

Cek outliers dari variabel weight menggunakan z score

In [15]:
df[(np.abs(stats.zscore(df["weight"]))>= 3)]

Empty DataFrame
Columns: [name, birth_year, height, weight, city, gender]
Index: []

Cek outliers dari variabel weight menggunakan IQR

In [16]:
q1 = df["weight"].quantile(0.25)
q3 = df["weight"].quantile(0.75)

iqr = q3-q1 #Interquartile range
fence_low  = q1-1.5*iqr
fence_high = q3+1.5*iqr

df.loc[(df["weight"] < fence_low) | (df["weight"] > fence_high)]

Empty DataFrame
Columns: [name, birth_year, height, weight, city, gender]
Index: []

Interpretasi 

Dengan pengecekan outliers dengan 2 metode, yaitu : zscore dan IQR. Didapatkan hasil bahwa pada data variabel weight tidak terdapat data outlier